### PR TITLE
Issue/287

### PR DESCRIPTION
**Pull request by minborg**
This PR removes most issues flagged as Bugs in SQ

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public static Method methodOrThrow(@NotNull final String className,
             return analyticsClass.getMethod(methodName, parameterTypes);
         } catch (ReflectiveOperationException e) {
             Jvm.rethrow(e);
-            return null;
+            throw new IllegalStateException("We should never end up here");
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java]: I wonder if it's possible/worth making something like `Jvm.rethrowable(ex)` so you can do `throw Jvm.rethrowable(ex)` and not need to do this?

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java]: Or `throw Jvm.toUnchecked(ex)` might be more explicit

File: src/main/java/net/openhft/chronicle/core/Jvm.java [hunk 2]
```diff
@@ -1606,6 +1611,8 @@ public static String getCpuClass() {
                             Jvm.warn().on(CpuClass.class, "process " + cmd + " returned " + ret);
                     } catch (InterruptedException e) {
                         Jvm.warn().on(CpuClass.class, "process " + cmd + " waitFor threw ", e);
+                        // Restore the interrupt state...
+                        Thread.currentThread().interrupt();
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/Jvm.java:1615]: nice

File: src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java]: yeah the standard pattern is `throw Jvm.rethrow(e)` - we should do that and not `throw new IllegalStateException("We should never end up here")` as that loses `e`

File: src/main/java/net/openhft/chronicle/core/io/BackgroundResourceReleaser.java [hunk 3]
```diff
@@ -98,7 +98,7 @@ public static void releasePendingResources() {
                     performRelease(o);
             }
             if (stopping)
-                RESOURCES.offer(POISON_PILL);
+                offerPoisonPill();
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/io/BackgroundResourceReleaser.java]: this changes behaviour. We don't want to warn if we can't add - I think these 2 changes should be reverted

File: src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java (hunk 1 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/internal/analytics/ReflectionUtil.java]: Fixed now

File: src/main/java/net/openhft/chronicle/core/io/BackgroundResourceReleaser.java (hunk 3 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/io/BackgroundResourceReleaser.java]: Added a flag for warning

**Review by JerryShea — APPROVED** [review]

**Pull request merged**